### PR TITLE
Merge visualization examples

### PR DESCRIPTION
**Pull request by hvy**
Fixes https://github.com/optuna/optuna/issues/604.

Note that changes other than merging existing examples were outside the scope of this PR.

### CODE REVIEW

File: examples/visualization/plot_study.ipynb [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import optuna\n",
+    "\n",
+    "optuna.logging.set_verbosity(optuna.logging.WARNING)  # This verbosity change is just to simplify the notebook output.\n",
+    "\n",
+    "study = optuna.create_study(direction='maximize', pruner=optuna.pruners.MedianPruner())\n",
+    "study.optimize(objective, n_trials=32)"
```
Comment (toshihikoyanase) [inline examples/visualization/plot_study.ipynb]: It takes about 30 mins to execute this cell in Google Colab.
Can we reduce the execution time by changing the size of the dataset like other examples?

File: examples/visualization/plot_study.ipynb [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    "    n_layers = trial.suggest_int('n_layers', 1, 3)\n",
+    "    for i in range(n_layers):\n",
+    "        layers.append(trial.suggest_int('n_units_l{}'.format(i), 32, 64))\n",
+    "\n",
+    "    clf = MLPClassifier(hidden_layer_sizes=tuple(layers))\n",
+    "\n",
+    "    for step in range(100):\n",
+    "        clf.partial_fit(x_train, y_train, classes=classes)\n",
+    "\n",
+    "        intermediate_value = 1.0 - clf.score(x_test, y_test)  # Report intermediate objective value.\n",
```
[the remark this answers is not in the document]
Comment (hvy) [reply examples/visualization/plot_study.ipynb]: Thanks, I fixed the error you pointed out and cleaned up the code a bit in https://github.com/optuna/optuna/pull/735/commits/bd019b2da01ac9931336214291ddf676aa512dfa.

File: examples/visualization/plot_study.ipynb [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from optuna.visualization import plot_parallel_coordinate\n",
+    "\n",
+    "plot_parallel_coordinate(study)"
```
[the remark this answers is not in the document]
Comment (hvy) [reply examples/visualization/plot_study.ipynb]: Changes the parameters to not be conditional, the lr and the number of neurons in each layer for a 3 layer MLP in https://github.com/optuna/optuna/pull/735/commits/bd019b2da01ac9931336214291ddf676aa512dfa.

File: examples/visualization/plot_study.ipynb (hunk 1 above)
Comment (hvy) [reply examples/visualization/plot_study.ipynb]: Changed to only use a subsample of 4k images (in contrast to previous 70k) in https://github.com/optuna/optuna/pull/735/commits/bd019b2da01ac9931336214291ddf676aa512dfa. Took roughly 5 min on my macbook.

**Comment by hvy** [comment]
Thanks for the comments, I addressed them in a single commit along with some cosmetics. Hope that was okay. PTAL.

File: examples/visualization/plot_study.ipynb [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+    "from sklearn.datasets import fetch_openml\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "mnist = fetch_openml(name='Fashion-MNIST', version=1)\n",
+    "classes = list(set(mnist.target))\n",
+    "\n",
+    "# For demonstrational purpose, only use a subset of the dataset.\n",
+    "n_samples = 4000\n",
+    "data = mnist.data[:n_samples]\n",
+    "target = mnist.target[:n_sample]\n",
```
Comment (crcrpar) [inline examples/visualization/plot_study.ipynb]: I'm not confident enough but, isn't this a typo?

```suggestion
    "target = mnist.target[:n_samples]\n",
```

Comment (hvy) [reply examples/visualization/plot_study.ipynb]: Sorry I should've verified. You're absolutely right.

File: examples/visualization/plot_study.ipynb [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+    "\n",
+    "def objective(trial):\n",
+    "    \n",
+    "    clf = MLPClassifier(\n",
+    "        hidden_layer_sizes=(\n",
+    "            trial.suggest_int('n_units_l0', 32, 64),\n",
+    "            trial.suggest_int('n_units_l1', 32, 64),\n",
+    "            trial.suggest_int('n_units_l2', 32, 64),\n",
+    "        ),\n",
+    "        learning_rate_init=trial.suggest_loguniform('learning_rate_init', 1e-5, 1e-1),\n",
```
Comment (toshihikoyanase) [inline examples/visualization/plot_study.ipynb]: `learning_rate_init` is a bit long and we can find the collision of labels in the contour plot as follows:
image
It may be environment-dependent, but can we shorten the name like `lr_init`?

File: examples/visualization/plot_study.ipynb [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+    "from sklearn.neural_network import MLPClassifier\n",
+    "\n",
+    "def objective(trial):\n",
+    "    \n",
+    "    clf = MLPClassifier(\n",
+    "        hidden_layer_sizes=(\n",
+    "            trial.suggest_int('n_units_l0', 32, 64),\n",
+    "            trial.suggest_int('n_units_l1', 32, 64),\n",
+    "            trial.suggest_int('n_units_l2', 32, 64),\n",
+    "        ),\n",
```
Comment (toshihikoyanase) [inline examples/visualization/plot_study.ipynb]: I think we can use a for-loop to suggest `n_units_l0`, `n_units_l1`, and `n_units_l2`, and it reduces the redundancy of code.

File: examples/visualization/plot_study.ipynb [hunk 7]
```diff
@@ -0,0 +1,272 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Visualizing High-dimensional Parameter Relationships\n",
+    "\n",
+    "This notebook demonstrates various visualizations of studies in Optuna.\n",
+    "The hyperparameters of a neural network trained to classify images are optimized and the resulting study is then visualized using these features.\n",
```
Comment (toshihikoyanase) [inline examples/visualization/plot_study.ipynb]: This document contains both `hyperparameter` and `hyper-parameter`. If you don't have a specific reason, we can use only `hyperparameter` like other documents.

File: examples/visualization/plot_study.ipynb (hunk 5 above)
Comment (hvy) [reply examples/visualization/plot_study.ipynb]: Ah I see, I changed the name to `lr_init` as you suggested.

File: examples/visualization/plot_study.ipynb (hunk 6 above)
Comment (hvy) [reply examples/visualization/plot_study.ipynb]: I actually thought that this was more comprehensible but okay I agree we can use some DRY. Fixed!

File: examples/visualization/plot_study.ipynb (hunk 7 above)
Comment (hvy) [reply examples/visualization/plot_study.ipynb]: Thank you for the catch. Just fixed it accordingly.

**Comment by hvy** [comment]
PTAL.

**Comment by toshihikoyanase** [comment]
I'm sorry, but I found one more point to be updated.
Please update a list of visualization examples in [`examples/README.md`](https://github.com/optuna/optuna/blob/master/examples/README.md).

**Comment by hvy** [comment]
Another catch, thank you. PTAL. 🙇

**Comment by toshihikoyanase** [comment]
Thank you! I'll merge this PR because it got two approvals.
